### PR TITLE
Add a simple example that utilizes the java api

### PR DESCRIPTION
**Pull request by abaranec**

### CODE REVIEW

**Comment by nbauernfeind** [comment]
@abaranec 

To fix the formatting errors causing the check failures, run:
```
./gradlew spotlessApply
```

File: java-client/weather-server-example/build.gradle [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//            '-XX:MaxGCPauseMillis=100',
+//            '-XX:+UseStringDeduplication' ]
+//}
+
+applicationDistribution.into('bin') {
+//    from(createGetSchemaScripts)
+//    from(createGetTsvScripts)
+//    from(createPollTsvScripts)
+//    from(createListTablesScripts)
+//    from(createExcessiveScript)
```
Comment (nbauernfeind) [inline java-client/weather-server-example/build.gradle:49]: Let's kill these commented lines.

File: java-client/weather-server-example/src/main/java/io/deephaven/client/examples/BarrageSupport.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.BitSet;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is here as a shim between the Java-API and the gRPC Barrage client.  This class will be superseded once
+ * proper support for retrieving Barrage tables directly is implemented, in the near future
```
Comment (nbauernfeind) [inline java-client/weather-server-example/src/main/java/io/deephaven/client/examples/BarrageSupport.java:22]: Suggesting this doc change:
```
 * proper support for retrieving Barrage tables directly is implemented. 
 * See issue deephaven-core#1262 to track this work.
```

File: java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/BarrageBackedTableModel.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
+        return false;
+    }
+
+    @Override
+    public Object getValueAt(int rowIndex, int columnIndex) {
+        // Typically you would access column data through a ColumnSource directly with get during iteration over an index
```
Comment (nbauernfeind) [inline java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/BarrageBackedTableModel.java:56]: Maybe:
```
// This is inefficient, but an efficient implementation is out of scope for this example. Typically,
// we want read data from ColumnSources through the chunking API. Here we'll be expedient
// and settle for the O(n log n) experience.

// Swing thinks about data in row positions, but the ColumnSource#get API uses row keys. Note
// how we convert from row position to row key using `Index#get`. 
```

File: java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/Ui.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+                    throw new RuntimeException("Channel not shutdown after 10 seconds");
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    {
+// GUI initializer generated by IntelliJ IDEA GUI Designer
```
Comment (nbauernfeind) [inline java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/Ui.java:178]: If IJ regenerates this on every execution, you probably want to mark the rest of this file as:
```
// @formatter:off
[ ... ]
// @formatter:on
```

File: java-client/weather-server-example/src/main/python/WeatherServer.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+def makeUpSomeData():
+    while True:
+        with(CITY_LOCK):
+            for lc in trackedCities:
+                tableWriter.logRow(DBTimeUtils.currentTime(), lc.state, lc.city, random.uniform(33,97), random.uniform(0,100))
+        time.sleep(1)
+
+bobby = Thread(target=periodicFetchRealData)
+bobby.start()
```
Comment (nbauernfeind) [inline java-client/weather-server-example/src/main/python/WeatherServer.py:196]: can you add an EOF newline if you don't mind?

File: java-client/weather-server-example/src/main/java/io/deephaven/client/examples/BarrageSupport.java (hunk 2 above)
Comment (abaranec) [reply java-client/weather-server-example/src/main/java/io/deephaven/client/examples/BarrageSupport.java:22]: Updated now that the comment is OBE

**Comment by devinrsmith** [comment]
recheck

**Comment by abaranec** [comment]
This was hanging around too long,  CI was not working properly. Will open another PR

**Comment by abaranec** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by abaranec** [comment]
recheck

**Comment by devinrsmith** [comment]
recheck

**Pull request closed without merging**